src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudIT.java

```diff
@@ -362,6 +369,51 @@ public void testTemplateNoGoogleLabels() throws Exception {
         }
     }
 
+    // Tests that no snapshot is created when we only have successful builds for given node
+    @Test(timeout = 300000, expected = GoogleJsonResponseException.class)
+    public void testNoSnapshotCreated() throws Exception {
+        logOutput.reset();
+
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(validInstanceConfiguration());
+
+        FreeStyleProject project = r.createFreeStyleProject();
+        Builder step = new Shell("echo works");
+        project.getBuildersList().add(step);
+        project.setAssignedLabel(new LabelAtom(LABEL));
+
+        FreeStyleBuild build = r.buildAndAssertSuccess(project);
+        Node worker = build.getBuiltOn();
+        worker.toComputer().doDoDelete();
+
+        assertFalse(logs(), logs().contains("snapshot"));
+        client.getSnapshot(projectId, worker.getNodeName());
```

Comment:
We should assert that result is null..

Reply:
The result does not return null for trying to retrieve a nonexistent snapshot.

Reply:
There should be way to check if snapshot was created or not..
If they are throwing exception then probably its easy to distinguish it from other exceptions..

Reply:
Agreed, having a direct means of checking if a snapshot exists would make this cleaner.

Reply:
https://cloud.google.com/compute/docs/reference/rest/v1/snapshots
I don't see any other means of checking if a snapshot exists. I also don't know how we can get more direct than trying to "get" the snapshot.
This is the result (the exception) that I receive from trying to get a nonexistent snapshot. I tried checking for a null or FAILED result, but the exception is thrown when I try to get this kind of result.

Reply:
One approach would be to have a client.isSnapshotExists() which returns true/false. To avoid having an exception thrown, the implementation could use this method: https://developers.google.com/resources/api-libraries/documentation/compute/v1/java/latest/com/google/api/services/compute/Compute.Snapshots.html#list-java.lang.String-

Reply:
The get method probably does something similar by going through a list of the snapshots. I agree the API isn't designed the way we wanted/expected, but I'm not sure this is necessary, especially for one test.

Reply:
What if `client.getSnapshot` handles the exception that occurs when a snapshot exists and returns null in that case, or rethrows the exception if the case wasn't a nonexistent snapshot?

Reply:
Changed so that if the exception occurs, we return null instead and now assertNull in the integration test.